cypress/e2e/ui/Automation/Embedded-Automate/Explorer/namespace.cy.js

```diff
@@ -0,0 +1,253 @@
+      'addNamespaceApi'
+    );
+    cy.get('#main-content .bx--btn-set button[type="submit"]')
+      .contains(addButton)
+      .click();
+    cy.wait('@addNamespaceApi');
+    cy.expect_flash(flashTypeSuccess, flashMessageAddSuccess);
+    // Selecting the created domain from the accordion list items
+    interceptAccordionApi(domainName);
+  });
+
+  it('Validate Cancel button', () => {
+    cy.toolbar(toolbarConfiguration, toolbarAddNewNamespace);
+    // Cancelling the form
+    cy.get('#main-content .bx--btn-set button[type="button"]')
+      .contains(cancelButton)
+      .should('be.enabled')
+      .click();
+    cy.expect_flash(flashTypeWarning, flashMessageCancelled);
+  });
+
+  it('Validate Name field allows only alphanumeric and _ . - $ characters', () => {
+    // Trying to add a namespace with invalid characters
+    addNamespace(invalidNamespaceName);
+    cy.expect_flash(flashTypeError, flashMessageInvalidNamespace);
+    // Cancelling the form
+    cy.get('#main-content .bx--btn-set button[type="button"]')
+      .contains(cancelButton)
+      .click();
+  });
+
+  it('Checking whether add, edit & delete namespace works', () => {
+    // Adding a new namespace
+    addNamespace();
+    cy.expect_flash(flashTypeSuccess, flashMessageAddSuccess);
+
+    // Selecting the created namespace from the accordion list items
+    interceptAccordionApi();
+    // Editing the namespace
+    cy.toolbar(toolbarConfiguration, toolbarEditNamespace);
+    // Checking if the Save button is disabled initially
+    cy.get('#main-content .bx--btn-set button[type="submit"]')
+      .contains(saveButton)
+      .should('be.disabled');
+    cy.get('#datastore-form-wrapper input#description')
+      .clear()
+      .type(editedDescription);
+    cy.get('#main-content .bx--btn-set button[type="submit"]')
+      .contains(saveButton)
+      .should('be.enabled')
+      .click();
+    cy.expect_flash(flashTypeSuccess, flashMessageSaveSuccess);
+
+    // Deleting the namespace
+    cy.expect_browser_confirm_with_text({
+      confirmTriggerFn: () =>
+        cy.toolbar(toolbarConfiguration, toolbarRemoveNamespace),
+      containsText: browserConfirmRemoveMessage,
+    });
+    cy.expect_flash(flashTypeSuccess, flashMessageNamespaceRemoved);
+  });
+
+  it('Checking whether creating a duplicate namespace is restricted', () => {
+    // Adding a new namespace
+    addNamespace();
+    // Trying to add duplicate namespace
+    addNamespace();
+    cy.expect_flash(flashTypeError, flashMessageNameAlreadyExists);
+
+    // Cancelling the form
+    cy.get('#main-content .bx--btn-set button[type="button"]')
+      .contains(cancelButton)
+      .click();
+  });
+
+  it('Checking whether Cancel & Reset buttons work fine in the Edit form', () => {
+    // Adding a new namespace
+    addNamespace();
+    // Selecting the created namespace from the accordion list items
+    interceptAccordionApi();
```

Comment:
I feel like it would be more intention revealing to not have a default for this function.  

`interceptAccordionApi(namespaceName);`